src/primitive/cuboid.rs

```diff
@@ -0,0 +1,222 @@
+use cgmath::{BaseFloat, Point3, Vector3};
+use cgmath::prelude::*;
+
+use {Aabb3, Ray3};
+use prelude::*;
+
+/// Cuboid primitive.
+///
+/// Have a cached set of corner points to speed up computation.
+#[derive(Debug, Clone, PartialEq)]
+#[cfg_attr(feature = "eders", derive(Serialize, Deserialize))]
+pub struct Cuboid<S> {
+    /// Dimensions of the box
+    pub dim: Vector3<S>,
+    half_dim: Vector3<S>,
+    corners: Vec<Point3<S>>,
+}
+
+impl<S> Cuboid<S>
+where
+    S: BaseFloat,
+{
+    /// Create a new rectangle primitive from component dimensions
+    pub fn new(dim_x: S, dim_y: S, dim_z: S) -> Self {
+        Self::new_impl(Vector3::new(dim_x, dim_y, dim_z))
+    }
+
+    /// Create a new rectangle primitive from a vector of component dimensions
+    pub fn new_impl(dim: Vector3<S>) -> Self {
+        Self {
+            dim,
+            half_dim: dim / (S::one() + S::one()),
+            corners: Self::generate_corners(&dim),
+        }
+    }
+
+    fn generate_corners(dimensions: &Vector3<S>) -> Vec<Point3<S>> {
+        let two = S::one() + S::one();
+        vec![
+            Point3::new(dimensions.x, dimensions.y, dimensions.z) / two,
+            Point3::new(-dimensions.x, dimensions.y, dimensions.z) / two,
+            Point3::new(-dimensions.x, -dimensions.y, dimensions.z) / two,
+            Point3::new(dimensions.x, -dimensions.y, dimensions.z) / two,
+            Point3::new(dimensions.x, dimensions.y, -dimensions.z) / two,
+            Point3::new(-dimensions.x, dimensions.y, -dimensions.z) / two,
+            Point3::new(-dimensions.x, -dimensions.y, -dimensions.z) / two,
+            Point3::new(dimensions.x, -dimensions.y, -dimensions.z) / two,
+        ]
+    }
+}
+
+impl<S> SupportFunction for Cuboid<S>
+where
+    S: BaseFloat,
+{
+    type Point = Point3<S>;
+
+    fn support_point<T>(&self, direction: &Vector3<S>, transform: &T) -> Point3<S>
+    where
+        T: Transform<Point3<S>>,
+    {
+        ::primitive::util::get_max_point(&self.corners, direction, transform)
```

Comment:
let's not use globally addressed functions and instead put some `use` on top of the file/function